File: algorithms/sort/bubble-sort/bubbleSort.js
const bubbleSort = arr => {
  let swapped

  do {
    swapped = false

    for (let i = 0; i < arr.length; i++) {
      if (arr[i] > arr[i + 1]) {
        [arr[i], arr[i + 1]] = [arr[i + 1], arr[i]]
        swapped = swapped || true
      }
    }
  } while (swapped)

  return arr
}

File: algorithms/sort/bubble-sort/bubble-sort.rb
def bubble_sort arr
  len = arr.length

  loop do
    swapped = false

    (len - 1).times do |i|
      if (arr[i] > arr[i + 1])
        arr[i], arr[i + 1] = arr[i + 1], arr[i]
        swapped = swapped || true
      end
    end

    break if not swapped
  end

  arr
end
